src/gui/processing/qgsprocessingwidgetwrapperimpl.cpp

```diff
@@ -4126,6 +4135,19 @@ void QgsProcessingPointPanel::setValue( const QgsPointXY &point, const QgsCoordi
   updateRubberBand();
 }
 
+void QgsProcessingPointPanel::showEvent( QShowEvent * )
+{
+  if ( mFirstShow )
+  {
+    // we don't support select on canvas if the dialog is modal
+    if ( QWidget *parentWindow = window() )
```

Comment:
I assume we do this in showEvent() since we can't get that information in the QgsProcessingPointPanel constructor? Probably the modality of the window might be changed atter QgsProcessingPointPanel has been created?

Reply:
That's correct -- it's not until after the dialog has been completely constructed, when a later call to "exec" actually causes the modal mode to kick in